### PR TITLE
Added support for managedFieldsManagers and Helm templating version

### LINKED ISSUE #396: Add support for application inputs
### Description

The provider is built without application fields:

[spec.source.helm.version](https://github.com/argoproj/argo-cd/blob/master/pkg/apis/application/v1alpha1/types.go#L106C40-L106C61)
[spec.ignore_difference.managed_fields_managers](https://github.com/argoproj/argo-cd/blob/master/pkg/apis/application/v1alpha1/types.go#L324)

image

### Potential Terraform Configuration

```hcl
resource "argocd_application" "main" {
  spec {
    source {
      helm {
        version = "v3"
        value_files = [
          "values.yaml"
        ]
      }
    }

    ignore_difference {
      group = "*"
      kind  = "*"
      json_pointers = [
        "/spec/replicas",
      ]
      managed_fields_managers = [
        "argocd-server"
      ]
    }
  }
}

```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by tonedefdev**
Closes #396

### CODE REVIEW

**Comment by onematchfox** [comment]
...

**Comment by tonedefdev** [comment]
> ...

Anything else needed @onematchfox to get this approved?

**Comment by onematchfox** [comment]
...

File: argocd/structure_application.go
Comment (onematchfox) [inline argocd/structure_application.go:314]: This code ensures that the version is correctly set when creating the application in ArgoCD however, you are missing the oppositye side of this code which ensures that the Terraform state is updated to reflect the current state of the application in ArgoCD. See [flattenApplicationSourceHelm](https://github.com/argoproj-labs/terraform-provider-argocd/blob/dfa0f72848fc1bb90f6ffe8cc9f943dd0f7c92e6/argocd/structure_application.go#L762).

File: argocd/structure_application.go
Comment (onematchfox) [inline argocd/structure_application.go:461]: Same issue here. See [flattenApplicationIgnoreDifferences](https://github.com/argoproj-labs/terraform-provider-argocd/blob/dfa0f72848fc1bb90f6ffe8cc9f943dd0f7c92e6/argocd/structure_application.go#L645C6-L645C41)

**Review by onematchfox — CHANGES_REQUESTED** [review]
Thanks for this PR and your patience in me getting to it. Please see comments inline. Also, please could you ensure that these fields are added to the tests:
- `version` can be added to [`TestAccArgoCDApplication_Helm`](https://github.com/argoproj-labs/terraform-provider-argocd/blob/dfa0f72848fc1bb90f6ffe8cc9f943dd0f7c92e6/argocd/resource_argocd_application_test.go#L96)
- `managed_field_managers` can be added to [`TestAccArgoCDApplication_IgnoreDifferences`](https://github.com/argoproj-labs/terraform-provider-argocd/blob/dfa0f72848fc1bb90f6ffe8cc9f943dd0f7c92e6/argocd/resource_argocd_application_test.go#L210)

File: argocd/resource_argocd_application_test.go
Comment (onematchfox) [inline argocd/resource_argocd_application_test.go:1691]:
``` suggestion
      json_pointers           = ["/spec/replicas"]
``` 

nit: mixed tabs/spaces (not your fault these tests are painful in this regard - spaces in *.go files :vomiting_face: )

File: argocd/resource_argocd_application_test.go
Comment (onematchfox) [inline argocd/resource_argocd_application_test.go:1706]:
``` suggestion
      managed_fields_managers = [
        "some-controller-owner",
        "some-other-controller-owner",
      ]
```

Good to check that everything works both with and without `json_pointers`

File: argocd/resource_argocd_application_test.go
Comment (onematchfox) [inline argocd/resource_argocd_application_test.go:293]:
``` suggestion
				ImportStateVerifyIgnore: []string{"wait", "cascade", "status", "validate"},
```

Need to add `validate` in here to fix tests after #411 was merged

**Review by onematchfox — CHANGES_REQUESTED** [review]
Couple final changes please :pray:

File: argocd/resource_argocd_application_test.go
Comment (tonedefdev) [reply argocd/resource_argocd_application_test.go:293]: I was wondering if this was what I needed to fix the tests! Thank you. I was struggling to figure out where `validate` was coming from. When I was building the provider on local I saw no trace of this, so thanks for clearing this up!

**Comment by tonedefdev** [comment]
> Couple final changes please 🙏

@onematchfox  - all tests are passing now and I made the requested changes! Hopefully we're good to merge now, but if we need more changes, I'm happy to oblige

**Review by onematchfox — APPROVED** [review]
LGTM - thanks for your patience and cooperation.

**Pull request merged**
